File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    // int findPivot(vector<int>& nums) {
    //     int low = 0, high = nums.size() - 1;

    //     while (low < high) {
    //         int mid = low + (high - low) / 2;
    //         if (nums[mid] > nums[high])
    //             low = mid + 1;
    //         else
    //             high = mid;
    //     }
    //     return low;
    // }

    bool check(vector<int>& nums) {
        int cnt = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] > nums[(i + 1) % nums.size()])
                cnt++;
            if (cnt > 1)
                return false;
        }
        return true;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.ts
function check(nums: number[]): boolean {
    let cnt = 0;
    for(let i = 0; i < nums.length; i++) {
        if(nums[i] > nums[(i+1)%nums.length]) cnt++;
        if(cnt > 1) return false;
    }
    return true;
};